reV/supply_curve/points.py

```diff
@@ -2,27 +2,1027 @@
+            Number of exclusion points per SC point along an axis.
+            This number**2 is the total number of exclusion points per
+            SC point.
+        exclusion_shape : tuple
+            Shape of the full exclusions extent (rows, cols). Inputing this
+            will speed things up considerably.
+        offshore_flags : np.ndarray | None
+            Array of offshore boolean flags if available from wind generation
+            data. None if offshore flag is not available.
+        close : bool
+            Flag to close object file handlers on exit
+
+        Returns
+        -------
+        ndarray
+            Exclusions weighted means of data for supply curve point
+        """
+        kwargs = {"excl_dict": excl_dict, "resolution": resolution,
+                  "exclusion_shape": exclusion_shape, "close": close}
+        with cls(gid, excl, tm_dset, **kwargs) as point:
+            means = point.exclusion_weighted_mean(data)
+
+        return means
+
+    @classmethod
+    def sc_sum(cls, gid, excl, tm_dset, data, excl_dict=None, resolution=64,
+               exclusion_shape=None, close=True):
+        """
+        Compute the aggregate (sum) of data for the sc point
+
+        Parameters
+        ----------
+        gid : int
+            gid for supply curve point to analyze.
+        excl : str | ExclusionMask
+            Filepath to exclusions h5 or ExclusionMask file handler.
+        tm_dset : str
+            Dataset name in the exclusions file containing the
+            exclusions-to-resource mapping data.
+        data : ndarray | ResourceDataset
+            Array of data or open dataset handler to apply exclusions too
+        excl_dict : dict | None
+            Dictionary of exclusion LayerMask arugments {layer: {kwarg: value}}
+            None if excl input is pre-initialized.
+        resolution : int
+            Number of exclusion points per SC point along an axis.
+            This number**2 is the total number of exclusion points per
+            SC point.
+        exclusion_shape : tuple
+            Shape of the full exclusions extent (rows, cols). Inputing this
+            will speed things up considerably.
+        offshore_flags : np.ndarray | None
+            Array of offshore boolean flags if available from wind generation
+            data. None if offshore flag is not available.
+        close : bool
+            Flag to close object file handlers on exit.
+
+        Returns
+        -------
+        ndarray
+            Sum / aggregation of data for supply curve point
+        """
+        kwargs = {"excl_dict": excl_dict, "resolution": resolution,
+                  "exclusion_shape": exclusion_shape, "close": close}
+        with cls(gid, excl, tm_dset, **kwargs) as point:
+            agg = point.aggregate(data)
+
+        return agg
+
+
+class AggregationSupplyCurvePoint(SupplyCurvePoint):
+    """Generic single SC point based on exclusions, resolution, and techmap"""
```

Comment:
Update docstring: `"""Generic single SC point to aggregate data from an h5 file."""`

Reply:
good catch